internal/security/bootstrapper/command/setupacl/command.go

```diff
@@ -0,0 +1,364 @@
+	if err != nil {
+		return fmt.Errorf("Failed to read response body to get leader: %w", err)
+	}
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		trimmedResp := strings.TrimSpace(string(responseBody))
+		// Consul's raft leader election process could take a bit of time
+		// before responds back with a non-empty leader
+		if len(trimmedResp) == 0 || emptyLeader == trimmedResp {
+			return errors.New("no leader yet")
+		}
+		// now we have a cluster raft leader
+		c.loggingClient.Infof("leader [%s] is elected", trimmedResp)
+		return nil
+
+	// almost unlikely for this case unless URL is incorrect
+	default:
+		return fmt.Errorf("get Consul leader request failed with status code= %d: %s", resp.StatusCode, string(responseBody))
+	}
+}
+
+func (c *cmd) getSecretStoreTokenFromFile() (string, error) {
+	trimmedFilePath := strings.TrimSpace(c.configuration.StageGate.Registry.ACL.SecretsAdminTokenPath)
+	if len(trimmedFilePath) == 0 {
+		return emptyToken, errors.New("required StageGate_Registry_SecretsAdminTokenPath from configuration is empty")
+	}
+
+	tokenFileAbsPath, err := filepath.Abs(trimmedFilePath)
+	if err != nil {
+		return emptyToken, fmt.Errorf("failed to convert tokenFile to absolute path %s: %v", trimmedFilePath, err)
+	}
+
+	// since the secretstore token is created by another service, secretstore-setup,
+	// so here we want to make sure we have the file
+	if exists := helper.CheckIfFileExists(tokenFileAbsPath); !exists {
+		return emptyToken, fmt.Errorf("secretstore token file %s not found", tokenFileAbsPath)
+	}
+
+	type SecretStoreToken struct {
+		Authentication struct {
+			Token string `json:"client_token"`
+		} `json:"auth"`
+	}
+
+	tokenJSONFile, err := os.Open(trimmedFilePath)
+
+	if err != nil {
+		return emptyToken, fmt.Errorf("failed to open secretstore token file: %v", err)
+	}
+
+	var tokenData SecretStoreToken
+	if err := json.NewDecoder(tokenJSONFile).Decode(&tokenData); err != nil {
+		return emptyToken, fmt.Errorf("failed to decode secretstore json token file: %v", err)
+	}
+
+	c.loggingClient.Infof("successfully retrieved secretstore management token from %s", trimmedFilePath)
+
+	return tokenData.Authentication.Token, nil
+}
+
+// configureConsulAccess is to enable the Consul config access to Vault via consul/config/access API
+// see the reference: https://www.vaultproject.io/api-docs/secret/consul#configure-access
+func (c *cmd) configureConsulAccess(secretStoreToken string, consulToken string) error {
+	configAccessURL := fmt.Sprintf("%s://%s:%d%s", c.configuration.SecretStore.Protocol,
+		c.configuration.SecretStore.Host, c.configuration.SecretStore.Port, consulConfigAccessVaultAPI)
+	_, err := url.Parse(configAccessURL)
+	if err != nil {
+		return fmt.Errorf("failed to parse config Access URL: %v", err)
+	}
+
+	c.loggingClient.Debugf("configAccessURL: %s", configAccessURL)
+
+	type ConfigAccess struct {
+		RegistryAddress string `json:"address"`
+		BootstrapToken  string `json:"token"`
```

Comment:
`BootstrapToken ` => `consulToken` since you have been using `BootstrapToken ` for the ACL token. or better yet `AccessToken`??

Reply:
it is the same token actually, so i can change it to BootstrapACLToken for field name and this is post body/payload by the way

Reply:
Right, I was confused with the SecretStoreToken... The parameter `consulToken` name should be consistent as `bootstrapACLToken` `